test/models/project_test.rb

```diff
@@ -16,19 +16,17 @@ def clone_repository(project)
     Project.create!(name: "hello", repository_url: url).token.wont_be_nil
   end
 
-  describe "#last_released_with_commit?" do
-    it "returns true if the last release had that commit" do
-      project.releases.create!(commit: "XYZ", author: author)
-      assert project.last_released_with_commit?("XYZ")
-    end
+  describe "#last_release_contains_commit?" do
+    let(:repository) { mock() }
 
-    it "returns false if the last release had a different commit" do
-      project.releases.create!(commit: "123", author: author)
-      assert !project.last_released_with_commit?("XYZ")
+    before do
+      project.stubs(:repository).returns(repository)
+      repository.stubs(:downstream_commit?).with('LAST', 'NEW').returns(true)
     end
 
-    it "returns false if there have been no releases" do
```

Comment:
can we keep this test? I'm pretty sure that's why there needs to be a guard around `last_release` in project.rb - in the case of the very first release `last_release` is nil